domains/zeto/internal/zeto/fungible/handler_lock.go

```diff
@@ -233,70 +208,92 @@ func (h *lockHandler) Prepare(ctx context.Context, tx *types.ParsedTransaction,
 		return nil, i18n.NewError(ctx, msgs.MsgErrorUnmarshalProvingRes, err)
 	}
 
-	data, err := common.EncodeTransactionData(ctx, req.Transaction, types.ZetoTransactionData_V0)
+	inputSize := common.GetInputSize(len(req.InputStates))
+	inputs, err := utxosFromInputStates(ctx, req.InputStates, inputSize)
 	if err != nil {
-		return nil, i18n.NewError(ctx, msgs.MsgErrorEncodeTxData, err)
+		return nil, err
 	}
-	LockParams := map[string]interface{}{
-		"utxos":    decodedTransfer.Inputs,
-		"proof":    common.EncodeProof(proofRes.Proof),
-		"delegate": params.Delegate,
-		"data":     data,
+
+	lockedInfo, err := parseInfoStatesJson(req.InfoStates[0].StateDataJson)
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorParseInfoStates, err)
 	}
-	paramsJSON, err := json.Marshal(LockParams)
+	unlockedOutputStates := req.OutputStates[:len(req.OutputStates)-int(lockedInfo.Size.Uint64())]
+	lockedOutputStates := req.OutputStates[len(req.OutputStates)-int(lockedInfo.Size.Uint64()):]
```

Comment:
Tracking the size of these slices feels a little bit fragile. Can we just look at the state data to separate them (either they are two different schemas, or they have a "locked" flag that tells whether they are a locked/unlocked state)?